ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -228,6 +231,57 @@ type TangConfig struct {
 	Thumbprint string `yaml:"thumbprint" json:"thp"`
 }
 
+type DiskPartition struct {
+	Device     string       `yaml:"device"`
+	Partitions []Partitions `yaml:"partitions"`
+}
+
+type Partitions struct {
+	MountPoint    string `yaml:"mount_point" `
+	Size          int    `yaml:"size"`
+	Start         int    `yaml:"start"`
+	MountFileName string `yaml:"-"`
+	Label         string `yaml:"-"`
+}
+
+type PartitionsUserInput Partitions
+
+func (prt *Partitions) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var userInput = PartitionsUserInput{}
+
+	err := unmarshal(&userInput)
+
+	if userInput.Start < 25000 {
+		// 25000 min value, otherwise root partition is too small and the installation will fail
+		err = fmt.Errorf("start value too small. must be over 25000")
+	}
+
+	if userInput.Size <= 0 {
+		err = fmt.Errorf("choose an appropritate disk size. must be greater than 0")
+	}
+
+	// it's a required field
+	if userInput.MountPoint == "" {
+		err = fmt.Errorf("must provide a path for mount_point. e.g /var/path")
+	}
+	// run a clean to ensure path is not malformed
+	userInput.MountPoint = path.Clean(userInput.MountPoint)
+	// ensure path is absolute
+	if !(path.IsAbs(userInput.MountPoint)) {
+		userInput.MountPoint = path.Join("/", userInput.MountPoint)
```

Comment:
If the Path isn't absolute and it's not clear what the intent is, is it safe to assume we're at the root or should we fail here since we don't know if the user made a mistake?